### PR TITLE
Add a schema for team member. Connects #54.

### PR DESCRIPTION
**Pull request by ertzeid**

### CODE REVIEW

File: lib/team_api/team_member_schema.json [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+        "description": "Brief descriptions of team member's interests",
+        "items": {
+          "type": "string"
+        },
+        "uniqueItems": true
+      },
+    },
+    "required": [
+      "name",
+      "full_name",
```
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: I know they can be derived from `full_name`, but I wonder if `first_name` and `last_name` should just be required as well.  @mbland what do you think?

Comment (mbland) [inline lib/team_api/team_member_schema.json]: Yeah. For example, Annalee's last name is "Flower Horne". Hard to parse that without an explicit `last_name` field.

File: lib/team_api/team_member_schema.json [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      "email",
+      "location"
```
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: I'm wondering about the `slack` and `github` fields being required as well, the thinking being that these are pretty critical to our communication and work.  Do we have folks without these things?  I can kind of see no GitHub account but even that's a bit tough here.

File: lib/team_api/team_member_schema.json [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    ]
+  }
+}
```
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: Looks like we need a newline for the EOF. :-)

File: lib/team_api/team_member_schema.json [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      },
+      "slack": {
+        "type": "string",
+        "description": "Team member's Slack username"
+      },
+      "resume": {
+        "type": "string",
+        "description": "URL for the team member's resume"
+      },
+      "okrs": {
```
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: Since this is meant to be a URL, should this property name be "okr_url" or something like that?

File: lib/team_api/team_member_schema.json [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+      },
+      "location": {
+        "type": "string",
+        "description": "Team member's location"
+      },
+      "slack": {
+        "type": "string",
+        "description": "Team member's Slack username"
+      },
+      "resume": {
```
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: Since this is meant to be a URL, should this property name be "resume_url" or something like that?

**Comment by ccostino** [comment]
Thanks, @ertzeid!  I have some questions and am curious to hear @mbland's thoughts, but this looks solid overall. :-)

File: lib/team_api/team_member_schema.json (hunk 2 above)
Comment (mbland) [inline lib/team_api/team_member_schema.json]: My main hesitation would be that other organizations wouldn't necessarily have these fields. Second-order schema validation for our specific use case may be called for. It'd be nice if it were pluggable.

File: lib/team_api/team_member_schema.json [hunk 6]
```diff
@@ -0,0 +1,84 @@
+{
+  "team": {
+    "title": "team - [team.member].yml files",
+    "description": "Team member metadata",
+    "$schema": "http://json-schema.org/draft-04/hyper-schema",
+    "type": "object",
+    "properties": {
+      "name": {
```
Comment (mbland) [inline lib/team_api/team_member_schema.json]: Tilting towards "id" here and everywhere else, though maybe in a future PR.

File: lib/team_api/team_member_schema.json [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+        "type": "string",
+        "description": "Unique name for the team member, derived from their GSA email address. Must match the file basename."
```
Comment (mbland) [inline lib/team_api/team_member_schema.json]: Would prefer to tilt away from hardcoding 18F/GSA-specific things in here, though perhaps it's a future PR issue.

File: lib/team_api/team_member_schema.json [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+      },
+      "github": {
+        "type": "string",
+        "description": "Team member's GitHub username"
+      },
+      "email": {
+        "type": "string",
+        "description": "Team member's GSA email address"
+      },
+      "location": {
```
Comment (mbland) [inline lib/team_api/team_member_schema.json]: Should probably specify that this should match one of the IDs from the `/locations` endpoint.

File: lib/team_api/team_member_schema.json [hunk 9]
```diff
@@ -0,0 +1,84 @@
...
+        "type": "string",
+        "description": "URL for the team member's resume"
+      },
+      "okrs": {
+        "type": "string",
+        "description": "URL for the team member's OKRs"
+      },
+      "start_date": {
+        "type": "date",
+        "description": "Date on which team member started at 18F"
```
Comment (mbland) [inline lib/team_api/team_member_schema.json]: Format? Hopefully YYYY-MM-DD (ISO 8601).

File: lib/team_api/team_member_schema.json [hunk 10]
```diff
@@ -0,0 +1,84 @@
...
+      },
+      "end_date": {
+        "type": "date",
+        "description": "Date on which team member left or is expected to leave"
```
Comment (mbland) [inline lib/team_api/team_member_schema.json]: YYYY-MM-DD

File: lib/team_api/team_member_schema.json (hunk 6 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: I don't want to both add a schema *and* change the standards all at the same time. I agree that ID might be more descriptive of what the field is used for, but that should be a separate issue.

File: lib/team_api/team_member_schema.json (hunk 7 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Done.

File: lib/team_api/team_member_schema.json (hunk 8 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Done.

File: lib/team_api/team_member_schema.json (hunk 5 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: I'm doing this all based on what I've observed. If we want to change the field names, I recommend doing that in separate Issues/PRs.

File: lib/team_api/team_member_schema.json (hunk 9 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Oops, meant to include that. Done.

File: lib/team_api/team_member_schema.json (hunk 10 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Done.

File: lib/team_api/team_member_schema.json (hunk 5 above)
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: Ah okay, makes sense!

File: lib/team_api/team_member_schema.json (hunk 1 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Done.

File: lib/team_api/team_member_schema.json (hunk 9 above)
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: +1 to ISO 8601 format.

File: lib/team_api/team_member_schema.json (hunk 3 above)
Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Huh. I based this file off of the other schema.json file from about_yml. I wonder if GitHub cares more than JSON parsers? Will add anyway.

File: lib/team_api/team_member_schema.json (hunk 2 above)
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: Oh good point, I hadn't considered that use case.

Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: I'll leave it optional for now. Maybe would be nice to add an option for overriding/extending the schema when implementing, but again, another Issue for another day.

Comment (ertzeid) [inline lib/team_api/team_member_schema.json]: Just pushed updated version, ready for perusal.

File: lib/team_api/team_member_schema.json (hunk 6 above)
Comment (ccostino) [inline lib/team_api/team_member_schema.json]: +1 to this; we should do this in a separate PR, and make sure to coordinate the change with anywhere else it might be needed.

**Comment by mbland** [comment]
LGTM. @ccostino?

**Comment by ccostino** [comment]
LGTM too, merging!
